src/index.ts

```diff
@@ -435,11 +436,87 @@ async function generateDocumentation(hre: HardhatRuntimeEnvironment): Promise<vo
       }
     };
 
-    const libraryNodes = AST.filter((node: any) => node.contractKind === 'library');
+    const parseNatspecOfErrorsFromAST = async () => {
+      for (let i = 0; i < AST.length; i += 1) {
+        if (AST[i].nodeType === 'ErrorDefinition') {
+          const errorName: string = AST[i].name;
+
+          const errorDocs: { tag?: string; description?: string }[] = AST[i].documentation?.text
+            .split('@')
+            .map((elem: string) => ({
+              tag: elem.substring(0, elem.indexOf(' ')),
+              description: elem.substring(elem.indexOf(' ') + 1),
+            }));
+
+          const errorParams: { paramType: string; paramName: string }[] = AST[i].parameters.parameters.map(
+            (elem: any) => ({
+              paramType: elem.typeName.name,
+              paramName: elem.name,
+            }),
+          );
+
+          const code = `${errorName}(${errorParams.map(
+            (elem: { paramType: string; paramName: string }) => elem.paramType,
+          )})`;
+          const notice = errorDocs?.filter(
+            (elem: { tag?: string; description?: string }) => elem.tag === 'notice',
+          )[0];
+          const dev = errorDocs?.filter(
+            (elem: { tag?: string; description?: string }) => elem.tag === 'dev',
+          )[0];
+          const custom = errorDocs?.filter((elem: { tag?: string; description?: string }) =>
+            elem.tag?.startsWith('custom'),
+          );
+
+          const inputs: { [key: string]: Param } = {};
+          errorParams.forEach((elem: { paramType: string; paramName: string }) => {
```

Comment:
Suggesting adding some blank lines for readability, as there are quite few nested `forEach` statements:

```js
errorParams.forEach((elem: { paramType: string; paramName: string }) => {
            
    let elemDescription: string = '';
            
    errorDocs?.forEach((docsElem: { tag?: string; description?: string }) => {
            // ...
    }
}
```